tests/next_tests/integration_tests/feature_tests/cases.py

```diff
@@ -0,0 +1,358 @@
+    """Initialize a field with a unique value in each coordinate."""
+    svals = tuple(sizes.values())
+    return embedded.np_as_located_field(*sizes.keys())(
+        np.arange(np.prod(svals), dtype=dtype).reshape(svals)
+    )
+
+
+def scalar5(dtype: str, shape: Optional[tuple[int, ...]]) -> int | float | np.array:
+    """Initialize a scalar to the value 5."""
+    if shape:
+        return np.ones(np.prod(shape), dtype=dtype) * 5
+    return np.dtype(dtype).type(5)
+
+
+@dataclasses.dataclass(frozen=True)
+class Builder:
+    partial: functools.partial
+
+    def build(self, *args, **kwargs) -> Self:
+        return self.partial(*args, **kwargs)
+
+    def __call__(self, *args, **kwargs) -> Any:
+        return self.build(*args, **kwargs)
+
+
+def make_builder(func):
+    """Create a fluid inteface for a function with many arguments."""
+
+    def make_setter(argname):
+        def setter(self, arg):
+            return self.__class__(
+                partial=functools.partial(
+                    self.partial, *self.partial.args, **(self.partial.keywords | {argname: arg})
+                )
+            )
+
+        setter.__name__ = argname
+        return setter
+
+    argspec = inspect.getfullargspec(func)
+
+    @dataclasses.dataclass(frozen=True)
+    class NewBuilder(Builder):
+        for argname in argspec.args + argspec.kwonlyargs:
+            locals()[argname] = make_setter(argname)
+
+    func_snake_words = func.__name__.split("_")
+    func_camel_name = "".join(word.capitalize() for word in func_snake_words)
+
+    NewBuilder.__name__ = f"{func_camel_name}Builder"
+
+    return NewBuilder(functools.partial(func))
+
+
+def allocate(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+    name: str,
+) -> Builder:
+    """Allocate a parameter or return value from a fieldview code object with a fluid interface."""
+    return (
+        make_builder(allocate_)
+        .sizes(case.default_sizes)
+        .case(case)
+        .fieldview_prog(fieldview_prog)
+        .name(name)
+    )
+
+
+def get_param_types(
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+) -> dict[str, ts.FieldType | ts.TupleType | ts.ScalarType]:
+    if fieldview_prog.definition is None:
+        raise ValueError(
+            f"test cases do not support {type(fieldview_prog)} with empty .definition attribute (as you would get from .as_program())!"
+        )
+    annotations = typing.get_type_hints(fieldview_prog.definition)
+    return {
+        name: type_translation.from_type_hint(type_hint) for name, type_hint in annotations.items()
+    }
+
+
+def extend_sizes(
+    sizes: dict[common.Dimension, str],
+    extend: Optional[dict[common.Dimension, tuple[int, int]]] = None,
+) -> dict[common.Dimension, str]:
+    sizes = sizes.copy()
+    if extend:
+        for dim, (lower, upper) in extend.items():
+            sizes[dim] += upper - lower
+    return sizes
+
+
+RETURN = "return"
+
+
+def allocate_(
```

Comment:
Just a question: why not `_allocate()`? If the builder is meant to be the only interface this one should be hidden, right? Or if both are needed for some corner cases, maybe we could use something more explicit like `allocate_field`?